pkg/pillar/zedUpload/datastore_aws.go

```diff
@@ -193,6 +196,35 @@ func (ep *AwsTransportMethod) processS3Download(req *DronaRequest) (error, int)
 	return err, csize
 }
 
+func (ep *AwsTransportMethod) processS3DownloadByChunks(req *DronaRequest) error {
+	pwd := strings.TrimSuffix(ep.apiKey, "\n")
+	dataChan := make(utils.ChunkDataChan)
+	go func(req *DronaRequest, dataNotif utils.ChunkDataChan) {
+		for {
+			select {
+			case data, ok := <-dataNotif:
+				if !ok {
+					return
+				}
+				ep.ctx.postChunk(req, data.Size, data.Chunk)
+			}
+		}
+	}(req, dataChan)
+
+	sc := zedAWS.NewAwsCtx(ep.token, pwd, ep.region, ep.hClient)
+	if sc == nil {
+		return fmt.Errorf("unable to create S3 context")
+	}
+
+	err := sc.DownloadFileByChunks(req.objloc, ep.bucket, req.name, dataChan)
+	if err != nil {
+		close(dataChan)
```

Comment:
Don't you also need a req.cleanUpByteData here? Some data could have been passed on the dataChan